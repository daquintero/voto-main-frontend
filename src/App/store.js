// Reducer Libraries
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { throttle } from 'lodash';
import { reducer as formReducer } from 'redux-form';

import mapReducer from '../Reusable/Map/redux/reducer';
import detailPageReducer from '../Assembly/DetailedPage/redux/reducer';
// TODO ADD LATER
// import homeReducer from '../Assembly/Social/redux/reducer';
import basicHomeReducer from '../Assembly/Home/redux/reducer';
import welcomeReducer from '../Layout/TopBar/redux/reducer';
import knowledgeBaseReducer from '../Assembly/KnowledgeBase/redux/reducer';
import reusableReducer from '../Reusable/redux/reducer';

// Local Storage
import { loadState, saveState } from './localStorage';


const reducer = combineReducers({
  openPage: detailPageReducer,
  map: mapReducer,
  home: basicHomeReducer,
  form: formReducer,
  welcome: welcomeReducer,
  knowledgeBase: knowledgeBaseReducer,
  reusable: reusableReducer,
});


const storedState = loadState();


const store = createStore(
  reducer,
  storedState,
  composeWithDevTools(applyMiddleware(thunk)),
);

store.subscribe(throttle(() => {
  saveState({
    welcome: store.getState().welcome,
  });
}, 3000));

export default store;
