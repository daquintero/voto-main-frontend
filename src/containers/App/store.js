import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

// Reusable Reducer Functions
import {
  informativeSnippetReducer,
} from '../Reusable/reusable-redux-index';

const reducer = combineReducers({
  informativeSnippets: informativeSnippetReducer,
});

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
