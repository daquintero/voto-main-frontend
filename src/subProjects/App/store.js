import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

// Reusable Reducer Functions
import {
  informativeSnippetReducer,
  individualReducer,
  corruptionCaseReducer,
  organizationReducer,
  financialItemReducer,
  mapReducer,
} from '../Reusable/reusableReduxIndex';

const openPage = combineReducers({
  informativeSnippet: informativeSnippetReducer,
  individual: individualReducer,
  corruptionCase: corruptionCaseReducer,
  organization: organizationReducer,
  financialItem: financialItemReducer,
  map: mapReducer,
});

const reducer = combineReducers({
  openPage,
});

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
