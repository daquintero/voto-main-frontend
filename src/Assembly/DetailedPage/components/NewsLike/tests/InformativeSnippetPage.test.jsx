// Test for InformativeSnippetPage Page Component
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import InformativeSnippetPage from '../INFPage';
import store from '../../../../../App/store';

describe('<InformativeSnippetPage />', () => {
  const initialState = store;
  const match = { params: { id: 1 } } ;

  it(' renders correctly without crashing', () => {
    const tree = renderer
      .create(<>
        <Provider store={initialState}>
          <InformativeSnippetPage match={match} />
        </Provider>
      </>)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
