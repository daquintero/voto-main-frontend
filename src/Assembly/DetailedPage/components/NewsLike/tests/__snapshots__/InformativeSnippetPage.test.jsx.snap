// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformativeSnippetPage />  renders correctly without crashing 1`] = `
<div
  className="load__card"
>
  <div
    className="load__icon-wrap"
  >
    <div
      className="load__icon load__card lds-ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
